tof/models.py

```diff
@@ -98,26 +75,69 @@ class Meta:
 
     name = models.CharField(_('Field name'), max_length=250, help_text=_('Name field'))
     title = models.CharField(_('User field name'), max_length=250, help_text=_("Name user's field"))
-    content_type = models.ForeignKey(ContentType, limit_choices_to=~Q(app_label='tof'), on_delete=models.CASCADE)
+    content_type = models.ForeignKey(ContentType, limit_choices_to=~Q(app_label='tof'), on_delete=models.CASCADE, related_name='translatablefields')
 
     def __str__(self):
         return f'{self.content_type.model}|{self.title}'
 
     def save(self, *args, **kwargs):
         super().save(*args, **kwargs)
-        prepare_cls_for_translate(self.content_type.model_class(), self)
+        self.add_translation_to_class()
 
     def delete(self, *args, **kwargs):
-        cls = self.content_type.model_class()
-        ct_pk = self.content_type.pk
-        name = self.name
         super().delete(*args, **kwargs)
+        self.remove_translation_from_class()
+
+    def __get__(self, instance, instance_cls):
+        return instance.get_translation(self.name)
+
+    def __set__(self, instance, value):
+        translation = vars(instance)[self.name] = instance.get_translation(self.name)
+        setattr(translation, translation.get_lang() if hasattr(instance, '_end_init') else '_origin', str(value))
 
-        restore_cls_after_translate(
-            cls,
-            name,
-            ContentType.objects.filter(translatablefield__isnull=False, id=ct_pk).exists(),
-        )
+    def __delete__(self, instance):
+        vars(self).pop(self.name, None)
+        instance._meta._field_tof.pop(self.id, None)  # pragma: no cover
+
+    def save_translation(self, instance):
+        val = instance.get_translation(self.name)
+        if val:
+            translation, _ = instance._translations.get_or_create(field=self, lang_id=val.get_lang(),)
+            translation.value = val
+            translation.save()
+
+    def add_translation_to_class(self, trans_mng=None):
+        cls = self.content_type.model_class()
+        if not issubclass(cls, TranslationFieldMixin):
+            cls.__bases__ = (TranslationFieldMixin, ) + cls.__bases__
+            cls._meta._field_tof = {}
+            if CHANGE_DEFAULT_MANAGER and not isinstance(cls._default_manager, TranslationManager):
+                origin = cls.objects
+                new_mng_cls = type(f'TranslationManager{cls.__name__}', (TranslationManager, type(origin)), {})
+                trans_mng = trans_mng or new_mng_cls()
+                trans_mng.contribute_to_class(cls, trans_mng.default_name)
+                cls._meta.default_manager_name = trans_mng.default_name
+                # FIXME
+                del cls.objects
+                trans_mng.contribute_to_class(cls, 'objects')
+                origin.contribute_to_class(cls, 'objects_origin')
+        setattr(cls, cls._meta._field_tof.setdefault(self.id, self).name, self)
```

Comment:
If you change _field_tof key, you must will change [expand_filter decorator](https://github.com/mom1/django-tof/blob/master/tof/decorators.py#L68) else you break filter feature

Reply:
i dont seen yet filter_decorator. but it's right now can be more easy too.